### DISCUSSION TITLE
Regarding negative cases in the luna16 dataset

### DISCUSSION THREAD

**Discussion by apourchot** [Project-MONAI/tutorials#1582] [Q&A]
Hi,

I've been going through the dataset split files from the luna16 tutorial and noticed there aren't any negative cases, i.e. scans without bounding boxes. I get why they might be left out in training, but they seem to be missing in the validation datasets too. Could you help me understand a couple of things?

1. When did this approach start? It's surprising that the original challenge creators would skip negative cases, but maybe I'm missing something. I looked over the nnDetection scripts for making the splits and didn't find anything about filtering out scans based on the number of bounding boxes. But, again, I might have missed something.

2. Can you confirm if the metrics on the tutorial page include or exclude negative cases? If they're excluded, it seems like the metrics might be off, most likely over-estimated?

Thanks in advance for any info you can share!

**Comment by KumoLiu** [comment]
Hi @Can-Zhao, could you please help share some comments on this discussion?
Thanks in advance!

**Comment by Can-Zhao** [comment]
Hi @apourchot 

Thank you for reaching out.
Yes, the original challenge creators skipped negative cases. (Update: The original challenge creators did not skip images without nodules.)

I do not think you will need to worry about negative samples. If I understand correctly, you were worrying that there is no negative sample, so the network tends to predict positive samples (boxes) everywhere. In detection, positive sample means a box. But in an image, most of the voxels are not associated to a box, and they are all negative samples. If during inference, a model predicts boxes everywhere on the image, e.g., 10000 boxes on an image, it will give a high false positive rate, and it is reflected in metrics.

Hope it helps. Please feel free to let me know if needed.

**Reply by apourchot** [reply]
Hi,

Thanks for the response.

I understand the training aspect of the detection model, specifically the rationale behind excluding scans without bounding boxes during training.

However, my concern lies in the evaluation phase. It appears that scans lacking bounding boxes are not subjected to inference, an aspect presumably unknown prior to inference. This could lead to metrics that fail to capture false positives in scans without bounding boxes.

You mentioned that the challenge organizers omitted negative cases. Does this also apply to evaluations across different data splits? If possible, could you direct me to where this is documented? I've been trying to locate this information without success.

As far as I'm aware, the evaluation script provided [here](https://luna16.grand-challenge.org/Evaluation/) includes a file that lists all series in the dataset, against which model predictions are compared to the ground truth. This file seems to encompass all series, not just those with bounding boxes.

Thanks for your input and any additional information you can provide.

**Reply by Can-Zhao** [reply]
I understand your concern. But when the metric consider false positive, it has nothing to do with whether an image contains a ground truth (GT) box. It considers whether a regular voxel is misclassified as a box, and whether a GT box is correctly identified. So it has nothing to do with whether an image contains box or not.
If during inference, a model predicts boxes everywhere on the image, e.g., 10000 boxes on an image, it will give a high false positive rate, and it is reflected in metrics.

**Reply by Can-Zhao** [reply]
Regarding to excluded data samples, in the evaluation script, there is a file called annotation/annotations_excluded.csv, it contains excluded samples, even including some small nodules. https://luna16.grand-challenge.org/Data/#:~:text=Annotations%20that%20are%20not%20included%20in%20the%20reference%20standard%20(non%2Dnodules%2C%20nodules%20%3C%203%20mm%2C%20and%20nodules%20annotated%20by%20only%201%20or%202%20radiologists)%C2%A0are%20referred%20as%20irrelevant%20findings.%20The%20list%20of%20irrelevant%20findings%20is%20provided%20inside%20the%20evaluation%20script%C2%A0(annotations_excluded.csv).

**Reply by apourchot** [reply]
Thank you once again for your response.

My main concern is that the metric **neglects false positives in scanners without bounding boxes**, as these scanners are excluded from the evaluation process. This oversight inevitably leads to inflated overall metrics, since **the model is only capable of generating false positives in series that lack bounding boxes**.

Regarding the annotations_excluded file, it seems to be separate from the series itself. It catalogues regions which, although not strictly nodules, are sufficiently suspicious to make it unreasonable to classify them as false positives when detected by the model. This is evident from the information in the link you provided:

> Annotations not included in the reference standard (non-nodules, nodules smaller than 3 mm, and nodules noted by only one or two radiologists) are labeled as irrelevant findings. The list of these irrelevant findings can be found in the evaluation script under annotations_excluded.csv.

Still, from our discussions, it appears that negative scans were deliberately removed. This intentional exclusion, I believe, skews the comparison with the previously established state of the art in the reported table, rendering it somewhat unfair.

I'm not pointing fingers here, I truly appreciate the Monai team's work. I'm just looking to confirm how these metrics were computed and reported.

**Reply by gabmis** [reply]
Hi :)
Wondering about this as well: why did you remove the negative scans from the evaluation set? can't find any trace of that in the grand challenge documentation.
Have a nice week-end!

**Reply by Can-Zhao** [reply]
I'm confused. You seem to have two concerns. One is suspecting the images without nodules deliberately removed during evaluation. The other one is thinking that lacking of healthy subjects will cause metric to inflate. To answer your first concern, we used the exact evaluation script from the challenge. Running this evaluation script did not use our data split, and it used the annotations.csv that the organizers provide https://github.com/Project-MONAI/tutorials/blob/3fa155bdc0b2fb7acaf3471ced96dddabbc73665/detection/run_luna16_offical_eval.sh#L28. To answer your second concern, I think your definition of positive/negative samples is not what the metrics trying to measure. For an image with 3 nodules, in your definition, it will be one positive sample; but the metrics will consider it as three positive samples and numerous of negative samples.

**Reply by Can-Zhao** [reply]
I felt I might have missed something here. I do not understand why using the original evaluation script and original annotations.csv, annotations_excluded.csv, and seriesuids.csv can be wrong. Regarding the data split json files, @dongyang0122 I was wondering if you remember how the data split json files were created from data split pkl file?

**Reply by Can-Zhao** [reply]
@apourchot You are right about original challenge creators did not skip negative cases. After I look into annotations_excluded.csv file, I realized this file did not exclude the subjects without nodules. It excludes the suspicious nodules. I'm sorry, I misunderstand how the challenge data were created. I did not find evidence that challenge creators skipped negative cases. For how data split files were created, let me find some help from nnDetection authors.  
Hi Michael, @mibaumgartner, I was wondering if you remember how the data split pkl file was generated? Thank you so much!

**Reply by mibaumgartner** [reply]
Hi, the data of LUNA16 is located in 10 directories which indicate the 10 folds. The splits file is simply a reflection of that folder structure where 9/10 were used for training and 1/10 was set as the validation fold (as in https://github.com/wentaozhu/DeepLung). No additional filtering of cases was applied from my/our side (each split [train+val] contains all of the images of LUNA16 i.e. 888 images). 

LUNA16 is a subset of LIDC, I would recommend to check the two papers for more specific details. From a quick glance, the LIDC paper is not quite clear about a lower boundary but mentions some soft upper boundaries.

Unfortunately, it is also not quite uncommon that datasets only have images with pathologies inside them. This certainly biases the results to some extend and does not allow for patient level evaluation. Nevertheless, for a pure object detection task, these datasets are still useful since we have an unknown number of objects which need to be localised quite precisely to be counted as correct.

**Reply by Can-Zhao** [reply]
Thank you so much Michael @mibaumgartner for helping check it! My apologies for involving you. Your insight has been invaluable.

**Comment by dongyang0122** [comment]
@apourchot Apologies for any misunderstanding caused. The initial dataset splits were generated using the `annotation.csv` file (from LUNA16) in conjunction with the data splits employed by nnDetection. Due to the fact that only pathological cases are enumerated in the annotation.csv file, healthy subjects were inadvertently omitted from our data splits. We will promptly revise the data list and update the relevant numbers in this tutorial to rectify this oversight. Thank you for bringing this to our attention!

**Comment by Can-Zhao** [comment]
Upon reviewing the feedback, we have initiated a thorough examination of our data processing procedures and found that the images without nodules were omitted in data split json files unintentionally. We are in consultation with my colleagues to gain a comprehensive understanding of every step involved in our data handling.

It is important to note that any errors in our data were not intentional. We are dedicated to rectifying any inaccuracies and implementing measures to prevent similar issues in the future. We will generate correct data split json files and recompute the metrics soon.

We value the feedback from our community and are grateful for the opportunity to improve. We welcome any further suggestions or insights that could assist us in this process.

**Comment by apourchot** [comment]
Thank you very much for acknowledging the problem. Looking forward to the updated splits and metrics :slightly_smiling_face:

**Comment by Thibescobar** [comment]
Hello @apourchot, 

_________ EDIT ________
I was wrong, actually, it is the opposite...
See details here: https://github.com/MIC-DKFZ/nnDetection/issues/272#issuecomment-2376542059
______ END EDIT ______

(Maybe related to my recent post here on the GitHub of nnDetection: MIC-DKFZ/nnDetection#272)

I observe something like you I think. 

When I include "healthy" patients, the FROC score is decreased. I was explaining this like you, but just by hand, based on the fact that with no object to detect, only FP are possible, thus pushing scores toward lower values for fixed FP/scan. 

But a colleague challenged this explanation saying that for example if we have a score of Se 80% @ 2FP/scan in average, if we add negative healthy patients, sensitivity will remain the same, and we could expect 2FP/scan in average still on the new patients. 

Sounds also good to me... So I am a bit lost since I definitly observe this phenomenum.

Do you have elements of answer or explanation of this phenomenum please. 

Thank you in advance. 

Best, 
Thibault
